R/scale-discrete-.r

```diff
@@ -132,6 +134,14 @@ ScaleDiscretePosition <- ggproto("ScaleDiscretePosition", ScaleDiscrete,
     }
   },
 
+  get_breaks = function(self, limits = self$get_limits()) {
+    if (self$warn && length(self$get_limits()) == 0) {
```

Comment:
Hmm... there are some problems with our clever plan. The obvious place to put these would be at the beginning and end of `train_df`, but this method is only called on non-positional scales. For positional scales `train` is called directly - further, it is called twice and if there are multiple scales due to facetting it is called for each of them... Not saying we shouldn't add the methods, but it doesn't solve the problem we are trying to solve...